doc/nrf/protocols/matter/end_product/last_fabric_removal_delegate.rst

```diff
@@ -0,0 +1,85 @@
+You can define a reaction on the last fabric removal and decide whether the device should remove non-volatile data, perform a factory reset, perform a reboot, or implement custom scenarios.
+
+Pre-defined last fabric removal behaviors
+=========================================
+
+There are 4 pre-defined reactions to the last fabric removal, available within the :ref:`matter_samples`.
+All behaviors are implemented as a fabric table module delegation and the chosen reaction is run as a callback on each fabric removal.
+
+For more information see the :file:`fabric_table_delegate.h` file which is located in the Matter samples common directory.
+
+You can choose one of the following reactions to the last fabric removal and instruct the device to:
+
+   * Do not react to the last fabric removal and leave the current state of the device.
+   * Remove the network credentials, and leave application-specific non-volatile data.
+   * Remove the network credentials, leave application-specific non-volatile data, and then perform the reboot.
+   * Remove all stored non-volatile data (network credentials and application-specific) and perform the reboot.
+   * Remove the network credentials, leave application-specific non-volatile data, and start advertising Bluetooth LE Matter service.
+
+To read more about reactions to the last fabric and learn how to set the related kconfigs see :ref:`ug_matter_device_advanced_kconfigs`.
+
+Custom last fabric removal delegate
+===================================
+
+In some cases, the pre-defined reactions to the last fabric removal can be not enough to cover all required activities.
+For example, only one pre-defined scenario assumes removing application-specific non-volatile data but requires also rebooting the device.
+If the pre-defined reactions do not meet your expectations, you can define a custom reaction to the last fabric removal.
+
+To implement a custom reaction you need to define a new delegation of the Fabric Table module.
+The Fabric Table module is defined in the Matter stack and is responsible for managing Matter fabrics to which the device belongs.
+The module stores a list of the delegations and each registered one will be called when the device leaves a Matter fabric.
+You can define multiple delegations, so first of all, you need to disable the pre-defined one.
+
+Disabling the pre-defined reaction to the last fabric removal
+-------------------------------------------------------------
+
+To disable the pre-defined reaction to the last fabric removal set :kconfig:option:`CONFIG_CHIP_LAST_FABRIC_REMOVED_NONE` to ``y``.
+
+Implementing the custom Fabric Table delegation
+-----------------------------------------------
+
+To implement the custom Fabric Table delegation you need to complete the following points:
+
+   * Include :file:`app/util/attribute-storage.h` from the Matter stack core.
```

Comment:
```suggestion
   * Include the :file:`app/util/attribute-storage.h` file from the Matter stack core.
```
Should these be numbered steps instead of a bullet point list?

Reply:
I thought about it, but the user can do them in a random way, but they must complete all the points. So it is not necessary to impose numeration. Unless the enumerated list is more readable as an instruction step-by-step here. What do you think?

Reply:
If they can be implemented in any order, then they can be bullets.